cranelift-codegen/src/isa/registers.rs

```diff
@@ -338,3 +329,21 @@ impl<'a> fmt::Display for DisplayRegUnit<'a> {
         }
     }
 }
+
+#[test]
+fn assert_sizes() {
+    use cranelift_codegen_shared::constants;
+    use std::mem::size_of;
+
+    // In these tests, size_of returns number of bytes: we actually want the number of bits, so
+    // multiply these by 8.
+    assert!(
+        (size_of::<RegClassMask>() * 8) <= constants::MAX_NUM_REG_CLASSES,
+        "need to bump MAX_NUM_REG_CLASSES or change RegClassMask type"
+    );
```

Comment:
Maybe use a const assertion (`const A: () = [()][(size_of::<RegClassMask>() * 8) > constants::MAX_NUM_REG_CLASSES) as usize]` When size_of > max num, then casting the bool to an usize would return 1, which is out of bounds of `[()]`.

Reply:
I just saw this crate in this week in rust: